raft/server.cc

```diff
@@ -568,7 +589,8 @@ future<add_entry_reply> server_impl::execute_modify_config(server_id from,
     if (from != _id && !_fsm->get_configuration().contains(from)) {
         // Do not accept entries from servers removed from the
         // configuration.
-        co_return add_entry_reply{not_a_leader{server_id{}}};
+        co_return add_entry_reply{transient_error{format("Modify config from {} was discarded since "
+                                                         "it is not part of the configuration", from), false, {}}};
```

Comment:
It seems that in this case, the error is neither transient, nor the resulting behavior is desirable: the server will just retry, and even without waiting since `delay = false`.